tests/integration/api/Auth/AuthenticateWithApiKeyTest.php

```diff
@@ -159,7 +159,7 @@ public function process(
             });
         });
 
-        $pipe = app('flarum.api.middleware');
+        $pipe = app('flarum.api.handler');
```

Comment:
Huh, this stuff actually requires leaving in the `afterResolving()` hack for the API stack. I'd like to avoid that. Any good idea on how to handle this differently here?

Reply:
This is in fact why I left the `afterResolving()` in, I couldn't figure out any good ways to get around this issue. I also don't know if any extensions possibly need to use the `afterResolving()` method for anything.

Reply:
As for extensions relying on it, that won't be the case, as the binding name's was changed. :wink: 
Any (known) use-cases should be covered by the extender.

I took a look at the test and think the hack is not really needed, so I will push a replacement to master.

Reply:
And done in 76f7d566b2b2fc2e675283a8d621de02a16c10b1. Please rebase on the latest master and get rid of the `afterResolving()` calls. Then we can (squash-)merge this. :raised_hands: